docs/guides/src/main/server/configuration.adoc

```diff
@@ -4,202 +4,237 @@
 
 <@tmpl.guide
 title="Configuring Keycloak"
-summary="An overview of the server configuration">
+summary="Understand how to configure and start Keycloak">
 
-This guide describes underlying core concepts of Keycloak configuration. It includes configuration guidelines for optimizing Keycloak for faster startup and low memory footprint.
+This guide explains the configuration methods for Keycloak and how to start and apply the preferred configuration. It includes configuration guidelines for optimizing Keycloak for faster startup and low memory footprint.
 
-== Server options
+== Configuration Sources for Keycloak
+Keycloak loads the configuration from four different configuration sources:
 
-You can view the server options by adding the `help` option for individual commands:
+* command-line parameters
+* environment variables
+* user-created `keycloak.conf` file
+* `keycloak.conf` file located in the `conf` directory.
 
-.Listing build options
-<@kc.build parameters="--help"/>
-
-.Listing configuration options
-<@kc.start parameters="--help"/>
-
-Alternatively, you can see the server options at <@links.server id="all-config"/>.
+Configuration sources have a descending ordinal: command-line parameters take precedence over environment variables. Environment variables take precedence over options set by using a specific configuration file. Options from a specific config file take precedence over options defined in `conf/keycloak.con`. When the same configuration key is found in multiple configuration sources, the used value is taken from the Source with the highest ordinal.
```

Comment:
Why "the Source" in capital letters?

Reply:
will be fixed in next update. also fixed: .con instead of .conf.